sync/requests.go

```diff
@@ -133,14 +163,20 @@ func ATxReqFactory(ids []types.AtxId) RequestFactory {
 				return
 			}
 
-			var tx []types.ActivationTx
-			err := types.BytesToInterface(msg, &tx)
+			var atxs []types.ActivationTx
+			err := types.BytesToInterface(msg, &atxs)
 			if err != nil {
 				s.Error("could not unmarshal atx data response: %v", err)
 				return
 			}
 
-			ch <- calcAndSetIds(tx)
+			atxs = calcAndSetIds(atxs)
+			if valid, err := validateAtxIds(ids, atxs); !valid {
```

Comment:
the error is defined in validateAtxIds

Reply:
The exact reason is from validateAtxIds but it is more clear to see that validation of atx ids failed.
Anyway, not critic.

Reply:
the validation is stated in the error the request code is relevant to the fetch flow
both strings will be printed it better this way in case more code will be added to validation and it would return different errors